templates/etc/marketplace.html

```diff
@@ -0,0 +1,94 @@
+    <label for="commission-class-preset">Type of Commission</label>
+      <select class="input last-input" name="pc" id="commission-class-preset">
+        <option value="" $:{is_selected(form.pc, "")}>Any/Other</option>
+        $for c in presets:
+          <option value="${c}" $:{is_selected(form.pc, c)}>${c}</option>
+      </select>
+      <h4 class="or color-lighter"><i>&mdash; Or &mdash;</i></h4>
+      <label for="commission-class">Other Commission Type</label>
+      <input id="commission-class" name="c" title="Commission Type"
+             type="text" tabindex="2" class="input" value="${form.c}">
+
+      <label for="commission-type-tags">Content</label>
+      <input id="commission-type-tags" name="q" title="Search for artists"
+             type="text" class="input" value="${form.q}">
+
+      <label for="commission-min-price">Minimum Price</label>
+      <input id="commission-min-price" name="min" title="Minimum Price"
+             type="text" class="input" value="${form.min}">
+      <label for="commission-max-price">Maximum Price</label>
+      <input id="commission-max-price" name="max" title="maximum Price"
+             type="text" class="input"value="${form.max}">
+      <label for="commish-currency">Currency</label>
+      <select class="input last-input" name="currency" id="commish-currency">
+        $for char, info in currencies.iteritems():
+          <option value="${char}" $:{is_selected(form.currency, char)}>${info['name']}</option>
+      </select>
+
+      <button class="button">Search</button>
+    </form>
+  </div>
+
+  <div class="sectioned-main content">
+    <h3>Artist Results</h3>
+
+    $if results:
+      $for artist in results:
+        <div class="clear marketplace-result">
+          <a href="/~${LOGIN(artist['username'])}" class="avatar">
+            $ avatar = artist['user_media']['avatar'][0]
+            <img src="${avatar['display_url']}" alt="" />
+          </a>
+          $# artist's commission info
+          <div class="artist-links">
+            <a href="/~${artist['username']}#user-commissions" class="username">${artist['username']}</a>
+            <a href="/submissions/${artist['username']}">[Gallery]</a>
+            $if artist['searchquery']:
+              <a href="/search?${artist['searchquery']}">[Tagged: ${artist['examplecount']}]</a>
+          </div>
+          $# commission price range
+          <div class="artist-prices">
+            $if artist['pricemin']:
+              <span>$:{'from ' if artist['pricemax'] else ''}$:{SYMBOL(artist['pricesettings'])} ${PRICE(artist['pricemin'])}</span>
+              $if artist['pricemax'] and artist['pricemax'] != artist['pricemin']:
+                <span>to $:{SYMBOL(artist['pricesettings'])} ${PRICE(artist['pricemax'])}</span>
+              $if artist['localmin'] != artist['pricemin']:
+                ( <span>Approx. $:{SYMBOL(form.currency)} ${PRICE(artist['localmin'])}</span>
```

Comment:
There shouldn’t be spaces inside the parentheses, even though web.py makes it awkward.

Maybe:

```html
$code:
  if artist['localmin'] != artist['pricemin']:
    local = 'Approx. %s %s' % (SYMBOL(form.currency), PRICE(artist['localmin']))
    if artist['pricemax'] and artist['pricemax'] != artist['pricemin']:
        local += ' to %s %s' % (SYMBOL(form.currency), PRICE(artist['localmax']))
(${local})
```